client/analytics/settings/config.js

```diff
@@ -25,27 +26,33 @@ const defaultOrderStatuses = [
 	'on-hold',
 ];
 
-const actionableOrderStatuses = Array.isArray(
-	wcSettings.wcAdminSettings.woocommerce_actionable_order_statuses
-)
-	? wcSettings.wcAdminSettings.woocommerce_actionable_order_statuses
+const {
+	woocommerce_actionable_order_statuses,
+	woocommerce_excluded_report_order_statuses,
+	woocommerce_default_date_range,
+} = getSetting( 'wcAdminSettings', {
+	woocommerce_actionable_order_statuses: [],
+	woocommerce_excluded_report_order_statuses: [],
+	woocommerce_default_date_range: DEFAULT_DATE_RANGE,
+} );
```

Comment:
Is there any case where `wcAdminSettings` might be defined but be an empty object or miss some of the properties? I imagine if `wcAdminSettings` is `{}`, none of the defaults would be applied here, right?

Reply:
Ya this was a tricky one because `wcAdminSettings` could have x number of props.  For the purpose of this library, the fallback defaults are sufficient because those are the only properties used.  I think this is another item that is ripe for further iteration in future refactors though.

So to answer your question, yes if `wcAdminSettings` fallback is `{}` then none of the default fallbacks would apply here.

Reply:
I'm ok with improving it in a further iteration. I guess in cases like this one we could just merge both objects, no?

Reply:
I'm not sure what you mean by "merge both objects".  The fallback is only used if the property doesn't exist in the settings state.  Another reason for having a fallback is _eventually_ we might load settings via api request vs a global from the server.  In which case having the fallback allows for what to use while the request is resolving.  Of course there'd be other changes needed if we go that route, but the interface for `getSetting` shouldn't need to change.

Reply:
I meant merging the fallback and the value if they are both objects. Something like this:
```diff
export function getSetting(
	name,
	fallback = false,
	validation = ( val ) => val
) {
	const value = allSettings.hasOwnProperty( name )
		? allSettings[ name ]
		: fallback;
+	if ( isPlainObject( fallback ) && isPlainObject( value ) ) {
+		return validation( { ...fallback, ...value } );
+	}
	return validation( value );
}
```
Another idea would be setting the fallback when destructuring:
```ES6
const {
	woocommerce_actionable_order_statuses = [],
	woocommerce_excluded_report_order_statuses = [],
	woocommerce_default_date_range = DEFAULT_DATE_RANGE,
} = getSetting( 'wcAdminSettings' );
```
This way, if the object exists but it has a missing property, it would still be assigned to the default. Do you think that would work?

Reply:
Ahh I see now.  Ya I don't think it's necessary to do merges here.  On the `get`, the calling code should only be concerned with the properties it needs via destructuring, so the fallback only needs to have those properties as well.  The fallback doesn't persist, otherwise we _would_ need something like you propose.